File: AppDFS.java
public class AppDFS
{
    public static void main(String[] args) {
        Graph g = new Graph("tinyG.txt");

        DepthFirstSearch dfs = new DepthFirstSearch(g, "0");        

        for(String v: g.getVerts()) {
            System.out.print(v+ ": ");  
            for(String w: dfs.pathTo(v)) {
                System.out.print(w+" ");
            }
            System.out.println();
        }
    }

}

File: appdfs.cpp
#include <iostream>
#include "graph.h"
#include "depthfirstsearch.h"

using namespace std;

int main()
{
  Graph g("tinyG.txt");
  DepthFirstSearch dfs(g, "0");

  for (auto const &v : g.getVerts())
  {
    cout << v << ": ";
    if (dfs.hasPathTo(v))
    {
      for (auto const &w : dfs.pathTo(v))
        cout << w << " ";
    }
    cout << endl;
  }
}
